File: rpc/dum-e-queue-producer.py
import json
import pika

connection = pika.BlockingConnection(pika.ConnectionParameters(
    host='localhost', port=5674
))
channel = connection.channel()

channel.queue_declare(queue='hello')

body = {
    'title': 'Hi!',
    'content': 'Hello world',
    'source': 'python dum-e producer'
}
channel.basic_publish(
    exchange='', body=json.dumps(body),
    routing_key='rideoctane-widget-generation',
)
channel.basic_publish(
    exchange='', body="Hellow world!",
    routing_key='rideoctane-widget-generation',
)
print(" [x] Sent 'Hello World!'")

connection.close()

File: dum-e-queue-producer.ts
var amqp = require('amqplib/callback_api');


amqp.connect('amqp://127.0.0.1:5674', function(connectionError, connection) {
    if (connectionError) {
        throw connectionError;
    }

    connection.createChannel(function(channelError, channel) {
        if (channelError) {
            throw channelError;
        }

        // ensure the queue exists
        const queue = 'rideoctane-widget-generation';
        channel.assertQueue(queue, {
            durable: false
        });

        let msg = {
            'title': 'blah',
            'content': 'Hello world',
            'source': 'typescript'
        };
        for(let i=0; i < 5; i++) {
            channel.sendToQueue(queue, Buffer.from(JSON.stringify(msg)));
            channel.sendToQueue(queue, Buffer.from("hellow world"));
            console.log(" [x] Sent %s", msg);
        }

        // exit after last message is published
        setTimeout(() => process.exit(0), 100)
    });
});
